tests/I-004-multi_stream_atomic_test.rs

```diff
@@ -0,0 +1,694 @@
+    };
+
+    let result = execute(&conflict_store, command, RetryPolicy::new()).await;
+
+    // Then: Source reflects debit, destination reflects injected audit between deposit and credit.
+    let from_events = conflict_store
+        .read_stream::<TestDomainEvents>(from_account.clone())
+        .await
+        .expect("reading source account stream succeeds after retry");
+    let to_events = conflict_store
+        .read_stream::<TestDomainEvents>(to_account.clone())
+        .await
+        .expect("reading destination account stream succeeds after retry");
+
+    let attempts = result
+        .as_ref()
+        .ok()
+        .and_then(|response| NonZeroU32::new(response.attempts()));
+    let actual = TransferAcceptanceResult {
+        succeeded: result.is_ok(),
+        attempts,
+        from_account: account_snapshot(&from_account, from_events.into_iter().collect()),
+        to_account: account_snapshot(&to_account, to_events.into_iter().collect()),
+    };
+
+    let expected = TransferAcceptanceResult {
+        succeeded: true,
+        attempts: Some(NonZeroU32::new(2).unwrap()),
+        from_account: account_snapshot(
+            &from_account,
+            vec![
+                TestDomainEvents::Credited {
+                    account_id: from_account.clone(),
+                    amount: from_initial_balance,
+                },
+                TestDomainEvents::Debited {
+                    account_id: from_account.clone(),
+                    amount: transfer_amount,
+                },
+            ],
+        ),
+        to_account: account_snapshot(
+            &to_account,
+            vec![
+                TestDomainEvents::Credited {
+                    account_id: to_account.clone(),
+                    amount: to_initial_balance,
+                },
+                TestDomainEvents::Audit {
+                    account_id: to_account.clone(),
+                },
+                TestDomainEvents::Credited {
+                    account_id: to_account.clone(),
+                    amount: transfer_amount,
+                },
+            ],
+        ),
+    };
+
+    assert_eq!(
+        actual, expected,
+        "retry logic should succeed after destination stream version conflict"
+    );
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct PartialVisibilityEvidence {
+    first_transfer_ok: bool,
+    first_attempts_at_least_one: bool,
+    second_transfer_ok: bool,
+    second_attempts_at_least_one: bool,
+    final_source_version: usize,
+    final_destination_version: usize,
+    final_source_balance: MoneyAmount,
+    final_destination_balance: MoneyAmount,
+    final_source_debits: Vec<u16>,
+    final_destination_transfer_credits: Vec<u16>,
+    event_counts_always_matched: bool,
+    debit_credit_counts_balanced: bool,
+}
+
+/// Scenario 3: Concurrent transfers never expose partially applied changes across streams.
+#[tokio::test]
+async fn concurrent_transfers_never_expose_partial_state() {
```

Comment:
I think this test looks right, and I can't really think of a simpler way to test this behavior, but I maybe we could put some better explanatory comments in the body so that it is easier to follow what and why?

Reply:
Agreed! Added comprehensive comments explaining the test strategy and the key invariant we're checking (debit/credit counts must always match across snapshots). The comments now explain:
- What we're testing (atomicity of multi-stream writes)
- How we're testing it (background poller + concurrent transfers)
- Why this proves atomicity (balanced debit/credit counts)

Fixed in cfd7458.

Reply:
Added inline comments throughout concurrent_transfers_never_expose_partial_state explaining each block (poller setup, concurrent execution, snapshot invariants, final assertions). Should be much easier to follow now.